storage/memory/memory_test.go

```diff
@@ -0,0 +1,92 @@
+package memory_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/Lubwama-Emmanuel/Interfaces/models"
+	"github.com/Lubwama-Emmanuel/Interfaces/storage/memory"
+	"github.com/Lubwama-Emmanuel/Interfaces/utils"
+)
+
+func TestMemory(t *testing.T) { //nolint:(gocognit)
+	t.Parallel()
+
+	type args struct {
+		number string
+		data   models.DataObject
+	}
+
+	tests := []struct {
+		testName string
+		args     args
+		wantErr  assert.ErrorAssertionFunc
+	}{
+		{
+			testName: "Success",
+			args: args{
+				number: "0706039119",
+				data: models.DataObject{
+					"0706039119": "Emmanuel",
+				},
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			testName: "Error",
+			args: args{
+				number: "2",
+				data:   models.DataObject{},
+			},
+			wantErr: assert.Error,
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.testName, func(t *testing.T) {
+			t.Parallel()
+
+			memoryDB := memory.NewMemoryStorage()
+
+			createErr := memoryDB.Create(tc.args.data)
+			if createErr != nil && tc.wantErr == nil {
+				assert.Fail(t, fmt.Sprintf("Test %v Error not expected but got one:\n"+"error: %q", tc.testName, createErr))
+				return
+			}
+
+			data, readErr := memoryDB.Read(tc.args.number)
+			if readErr != nil && tc.wantErr == nil {
+				assert.Fail(t, fmt.Sprintf("Test %v Error not expected but got one:\n"+"error: %q", tc.testName, readErr))
+				return
+			}
+
+			if !utils.CompareMaps(data, tc.args.data) {
```

Comment:
The `assert` library has methods for such a comparison such that you do not have to build your own